src/tests/ftest/deployment/disk_failure.py

```diff
@@ -143,8 +144,17 @@ def test_disk_fault_to_normal(self):
                 self.log.info("  %s: %s", key, device[key])
             try:
                 self.dmg_command.hostlist = NodeSet(host)
-                get_dmg_response(
-                    self, self.dmg_command.storage_replace_nvme,
-                    old_uuid=device["uuid"], new_uuid=device["uuid"])
+                # Set the device as faulty
+                self.dmg_command.json.value = True
+                result = self.dmg_command.storage_set_faulty(uuid=device["uuid"])
+                data = json.loads(result.stdout_text)
+                self.dmg_command.json.value = False
+                if data["error"]:
+                    self.fail("Test failed to set the NVMe faulty bit")
+                else:
+                    # Replace the device with same uuid.
+                    get_dmg_response(
+                        self, self.dmg_command.storage_replace_nvme,
+                        old_uuid=device["uuid"], new_uuid=device["uuid"])
```

Comment:
If we update `storage_set_faulty()` to use `self._get_json_result()` instead of `self._get_result()`, then this can be:
```suggestion
                # Set the device as faulty
                get_dmg_response(
                    self, self.dmg_command.storage_set_faulty, uuid=device["uuid"])

                # Replace the device with same uuid.
                get_dmg_response(
                    self, self.dmg_command.storage_replace_nvme,
                    old_uuid=device["uuid"], new_uuid=device["uuid"])
```

We can then also remove the `"storage_set_faulty"` `DmgCommand.METHOD_REGEX` entry.

The only other code that would need updating is `VmdLedStatus.set_device_faulty()` which would require this update:
```
    def set_device_faulty(self, device_id=None):
        """Get a device to faulty state.

        Args:
            device_id (str, optional): Device UUID. Defaults to None.
        Returns:
            dict: dmg device faulty information.
        """
        if device_id is None:
            self.fail("No device id provided")

        return get_dmg_response(
            self, self.dmg_command.storage_set_faulty, uuid=device_id)
```

Reply:
Done.